tests/framework/builder.py

```diff
@@ -202,24 +220,44 @@ def create(self,
                target_version: str = None,
                mem_file_name: str = "vm.mem",
                snapshot_name: str = "vm.vmstate",
-               net_ifaces=None):
+               net_ifaces=None,
+               use_ramdisk=False):
         """Create a Snapshot object from a microvm and artifacts."""
+        if use_ramdisk:
+            snaps_dir = self._microvm.jailer.chroot_ramfs_path()
+            mem_full_path = os.path.join(snaps_dir, mem_file_name)
+            vmstate_full_path = os.path.join(snaps_dir, snapshot_name)
+
+            # Pre-allocate ram for memfile to eliminate allocation variability.
+            utils.run_cmd(
+                'dd if=/dev/zero of={} bs=1M count=512'.format(mem_full_path)
+            )
+            cmd = 'chown {}:{} {}'.format(
```

Comment:
nit: `chown` calls identical to this one are sprinkled in multiple places over the test framework. We could have a method on the `Jailer` class for this called `chown_path(full_path)`

Reply:
sounds good to me